api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -252,5 +256,85 @@ else if (!isObsGroup) {
 			}
 		}
 	}
+
+	/**
+	 * This method validates Obs' concept values.
+	 * 
+	 * <ol>
+	 *     <li>Validates if high absolute and low absolute are within the valid range</li>
+	 *     <li>Validates if patient's age is within the valid range</li>
+	 * <ol/>
+	 * in the {@link ConceptReferenceRange}.
+	 *
+	 * @param obs Observation to validate
+	 * @param errors Errors to record validation issues
+	 *               
+	 * @since 2.7.0
+	 */
+	private void validateConceptReferenceRange(Obs obs, Errors errors) {
+		Concept concept = obs.getConcept();
+
+		if (concept != null && concept.getDatatype() != null && concept.getDatatype().isNumeric()) {
+			List<ConceptReferenceRange> crrList = Context.getConceptService()
+				.getConceptReferenceRangesByConceptId(concept.getConceptId());
+
+			if (!crrList.isEmpty()) {
+				ConceptReferenceRangeUtility utility = new ConceptReferenceRangeUtility();
+				boolean criteriaEvaluated = false;
+
+				for (ConceptReferenceRange referenceRange : crrList) {
+					if (utility.evaluateCriteria(referenceRange.getCriteria(), obs.getPerson())) {
```

Comment:
Did you look at that section of the talk post thread where we discussed the possibility of more than one criteria evaluating to true and how to choose the winner?

Reply:
You mean if 2 criteria are evaluating to true we should pick the criteria that has the most attributes right? That's at least what I got from the thread

Reply:
What do you mean by most attributes?

Reply:
> If we have a Male patient aged 20: They match both group 2 and 3 but group 3 has more factors so we use that one.

Did you mean to say that group 2 has more factors and hence we use it?

Reply:
Counting the number of factors to see which one has more, looks a bit more complicated because we delegate criteria evaluation fully to the velocity engine. My understanding from the talk thread discussion was that if more than one criteria evaluates to true, we use the one that has the most restrictive values.
For instance, in your example of a Male patient aged 20, if the `Male` criteria has a `hiAbsolute` of 20 and the `Male + age 10-30` criteria has a `hiAbsolute` of 30, we use the `Male` criteria because it has a more restrictive values. Did you interpret the discussion differently?

Reply:
I see. I had interpreted the "restriction" to be the factors in the criteria. As you've said, since we leave the criteria evaluation to velocity, it makes sense to use absolute values in this case. Thanks